compiler/rustc_middle/src/query/mod.rs

```diff
@@ -1462,10 +1462,13 @@ rustc_queries! {
     query missing_lang_items(_: CrateNum) -> &'tcx [LangItem] {
         desc { "calculating the missing lang items in a crate" }
     }
-    query visible_parent_map(_: ()) -> DefIdMap<DefId> {
-        storage(ArenaCacheSelector<'tcx>)
```

Comment:
I'm not sure what `ArenaCacheSelector` does here, should I use it instead of `Lrc<_>`?

Reply:
There was a performance hit before https://github.com/rust-lang/rust/pull/89395 and https://github.com/rust-lang/rust/pull/89120 were merged. The arena change may no longer be necessary.

Reply:
@Mark-Simulacrum  @jyn514 Now that benchmark results show small regression, should I revert this back (to remove `ArenaCacheSelector`)?

Reply:
That should still be done in a separate PR, so we can independently evaluate its effects.

Reply:
@Mark-Simulacrum So, is this discussion resolved then?

Reply:
Well, I'd expect that separate PR to likely get filed before this merges (in order to attempt to prevent the regression here).

Reply:
@Mark-Simulacrum I opened #92169 #92168 for that

Reply:
and #92170